lore/estimators/keras.py

```diff
@@ -177,63 +177,99 @@ def build_inputs(self):
+                if isinstance(encoder, Continuous):
+                    embedding = Dense(embed_size, activation='relu', name=embed_name)
+                else:
+                    embedding = Embedding(encoder.cardinality(), embed_size, name=embed_name)
 
-            if hasattr(encoder, 'sequence_length'):
-                embeddings[embed_name] = self.build_sequence_embedding(encoder, embedding, inputs, embed_name)
-            else:
-                embeddings[embed_name] = Reshape(target_shape=(embed_size,))(embedding(inputs[encoder.name]))
+                if hasattr(encoder, 'sequence_length'):
+                    embeddings[embed_name], layer = self.build_sequence_embedding(encoder, embedding, inputs, embed_name)
+                    if encoder.twin:
+                        embeddings[embed_name_twin], _ = self.build_sequence_embedding(encoder, embedding, inputs, embed_name, suffix='_twin', layer = layer)
+                else:
+                    embeddings[embed_name] = Reshape(target_shape=(embed_size,))(embedding(inputs[encoder.name]))
+                    if encoder.twin:
+                        embeddings[embed_name_twin] = Reshape(target_shape=(embed_size,))(embedding(inputs[encoder.twin_name]))
 
         return Concatenate()(list(embeddings.values()))
-    
-    def build_sequence_embedding(self, encoder, embedding, inputs, embed_name, suffix=''):
+
+    def build_sequence_embedding(self, encoder, embedding, inputs, embed_name, suffix='', layer = None):
         embed_size = encoder.embed_scale * self.embed_size
-    
+        raw_layer = None
+
         sequence = []
         for i in range(encoder.sequence_length):
             sequence.append(embedding(inputs[encoder.sequence_name(i, suffix)]))
+
         embed_sequence = Concatenate(name=embed_name + '_sequence' + suffix)(sequence)
     
         if self.sequence_embedding == 'flatten':
-            embedding = Flatten(name=embed_name + '_flatten' + suffix)(embed_sequence)
+            if layer == None:
+                flatten = Flatten
+                embedding = flatten(name=embed_name + '_flatten' + suffix)(embed_sequence)
+                raw_layer = flatten
+            else:
+                embedding = layer(name=embed_name + '_flatten' + suffix)(embed_sequence)
         else:
             sequence_embed_size = encoder.embed_scale * self.sequence_embed_size
             shaped_sequence = Reshape(target_shape=(encoder.sequence_length, embed_size))(embed_sequence)
             if self.sequence_embedding == 'lstm':
-                lstm = LSTM
-                if self.cudnn:
-                    if available_gpus > 0:
-                        lstm = CuDNNLSTM
-                    else:
-                        raise ValueError('Your estimator self.cuddn is True, but there are no GPUs available to tensorflow')
-                embedding = lstm(sequence_embed_size, name=embed_name + '_lstm' + suffix)(shaped_sequence)
+                if layer == None:
```

Comment:
Since you've added `raw_layer` can you pull the application out of each conditional, and have 1 line that applies `embedding = layer(sequence_embed_size, name=embed_name + '_' + self.sequence_embedding + suffix)(shaped_sequence)` before returning it?

Reply:
Cleaned up that method quite a bit.